--- K40_Whisperer-0.56_src/k40_whisperer.py	2021-01-24 13:33:20.000000000 -0500
+++ k40_whisperer.py	2021-01-25 14:22:40.000000000 -0500
@@ -5744,10 +5744,10 @@
 try:
     try:
         import tkFont
-        default_font = tkFont.nametofont("TkDefaultFont")
+        default_font = tkFont.nametofont("systemSystemFont")
     except:
         import tkinter.font
-        default_font = tkinter.font.nametofont("TkDefaultFont")
+        default_font = tkinter.font.nametofont("systemSystemFont")
 
     default_font.configure(size=9)
     default_font.configure(family='arial')
