--- K40_Whisperer-0.54_src/k40_whisperer.py	2020-10-11 13:02:06.000000000 -0700
+++ k40_whisperer.py	2021-01-25 06:43:30.000000000 -0800
@@ -107,7 +107,35 @@
     pass
 
 QUIET = False
-   
+
+# macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+# macOS Mojave and tikinter buttons are blank
+# https://stackoverflow.com/questions/52529403/button-text-of-tkinter-not-works-in-mojave]
+# Essentially the fix is to slightly resize the window after it opens.
+macOS_button_fix_enabled = False
+
+def macOS_button_fix(win):
+    def make_window_resizer(w):
+        def window_resizer():
+            a = w.winfo_geometry().split('+')[0]
+            (width, height) = a.split('x')
+            w.geometry('%dx%d' % (int(width)+1, int(height)+1))
+
+        return window_resizer
+
+    # The fix causes a bit of flicker on startup, so only run it for macOS >= 10.14
+    # Check for macOS >= 10.14
+    if macOS_button_fix_enabled:
+        try:
+            import platform
+            v, _, _ = platform.mac_ver()
+            v = float('.'.join(v.split('.')[:2]))
+            if v >= 10.14:
+                win.update()
+                win.after(0, make_window_resizer(win))
+        except:
+            pass
+ 
 ################################################################################
 class Application(Frame):
     def __init__(self, master):
@@ -573,6 +601,14 @@
         
         self.Label_GoToX   = Label(self.master,text="X", anchor=CENTER )
         self.Label_GoToY   = Label(self.master,text="Y", anchor=CENTER )
+
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        # Adjust button wrap locations for macOS
+        self.Open_Button.config(wraplength=20)
+        self.Reload_Button.config(wraplength=20)
+        self.Reng_Button.config(text="Raster Eng.")
+        self.Veng_Button.config(text="Vector Eng.")
+
         ###########################################################################
         # End Left Column #
 
@@ -3831,7 +3867,10 @@
                                   U_display))
 
         self.statusbar.configure( bg = 'white' )
-        
+
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        macOS_button_fix(root)
+         
     def menu_Inside_First_Callback(self, varName, index, mode):
         if self.GcodeData.ecoords != []:
             if self.VcutData.sorted == True:
@@ -4750,7 +4789,8 @@
         xd_label_L = 12
 
         w_label=150
-        w_entry=40
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        w_entry=100
         w_units=45
         xd_entry_L=xd_label_L+w_label+10
         xd_units_L=xd_entry_L+w_entry+5
@@ -4980,6 +5020,9 @@
         xd_entry_L=xd_label_L+w_label+10
         xd_units_L=xd_entry_L+w_entry+5
 
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        macOS_button_fix(gen_settings)
+
 
 
         D_Yloc=D_Yloc+D_dY
@@ -5306,7 +5349,8 @@
         xd_label_L = 12
 
         w_label=150
-        w_entry=40
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        w_entry=50
         w_units=35
         xd_entry_L=xd_label_L+w_label+10
         xd_units_L=xd_entry_L+w_entry+5
@@ -5352,6 +5396,9 @@
             
         self.EGV_Send = Button(egv_send,text="Send EGV Data",command=Close_and_Send_Click)
         self.EGV_Send.place(x=Xbut, y=Ybut, width=130, height=30, anchor="w")
+
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        macOS_button_fix(gen_settings)
         ################################################################################
         
         
@@ -5449,6 +5496,8 @@
 
     def apply(self):
         self.result = self.uom.get()
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        macOS_button_fix(gen_settings)
         return 
 
 
@@ -5790,4 +5839,7 @@
         app.master.geometry("480x320")
 
 
+# macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+macOS_button_fix_enabled = True
+macOS_button_fix(root)
 root.mainloop()
