--- K40_Whisperer-0.35_src/k40_whisperer.py	2019-08-25 22:01:44.000000000 -0400
+++ k40_whisperer.py	2019-09-08 12:37:02.000000000 -0400
@@ -93,8 +93,33 @@
 
 QUIET = False
 
-
-    
+# macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+# macOS Mojave and tikinter buttons are blank
+# https://stackoverflow.com/questions/52529403/button-text-of-tkinter-not-works-in-mojave]
+# Essentially the fix is to slightly resize the window after it opens.
+macOS_button_fix_enabled = False
+
+def macOS_button_fix(win):
+    def make_window_resizer(w):
+        def window_resizer():
+            a = w.winfo_geometry().split('+')[0]
+            (width, height) = a.split('x')
+            w.geometry('%dx%d' % (int(width)+1, int(height)+1))
+
+        return window_resizer
+
+    # The fix causes a bit of flicker on startup, so only run it for macOS >= 10.14
+    # Check for macOS >= 10.14
+    if macOS_button_fix_enabled:
+        try:
+            import platform
+            v, _, _ = platform.mac_ver()
+            v = float('.'.join(v.split('.')[:2]))
+            if v >= 10.14:
+                win.update()
+                win.after(0, make_window_resizer(win))
+        except:
+            pass
     
 ################################################################################
 class Application(Frame):
@@ -493,6 +518,14 @@
         
         self.Label_GoToX   = Label(self.master,text="X", anchor=CENTER )
         self.Label_GoToY   = Label(self.master,text="Y", anchor=CENTER )
+
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        # Adjust button wrap locations for macOS
+        self.Open_Button.config(wraplength=20)
+        self.Reload_Button.config(wraplength=20)
+        self.Reng_Button.config(text="Raster Eng.")
+        self.Veng_Button.config(text="Vector Eng.")
+
         ###########################################################################
         # End Left Column #
 
@@ -3384,7 +3417,10 @@
                                   U_display))
 
         self.statusbar.configure( bg = 'white' )
-        
+
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        macOS_button_fix(root)
+   
     def menu_Mode_Change_Callback(self, varName, index, mode):
         self.menu_View_Refresh()
 
@@ -4173,7 +4209,8 @@
         xd_label_L = 12
 
         w_label=150
-        w_entry=40
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        w_entry=100
         w_units=45
         xd_entry_L=xd_label_L+w_label+10
         xd_units_L=xd_entry_L+w_entry+5
@@ -4315,6 +4352,9 @@
         self.GEN_Close.place(x=Xbut, y=Ybut, width=130, height=30, anchor="center")
         self.GEN_Close.bind("<ButtonRelease-1>", self.Close_Current_Window_Click)
 
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        macOS_button_fix(gen_settings)
+
 
     ################################################################################
     #                          Raster Settings Window                              #
@@ -4586,7 +4626,8 @@
         xd_label_L = 12
 
         w_label=150
-        w_entry=40
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        w_entry=50
         w_units=35
         xd_entry_L=xd_label_L+w_label+10
         xd_units_L=xd_entry_L+w_entry+5
@@ -4632,6 +4673,9 @@
             
         self.EGV_Send = Button(egv_send,text="Send EGV Data",command=Close_and_Send_Click)
         self.EGV_Send.place(x=Xbut, y=Ybut, width=130, height=30, anchor="w")
+
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        macOS_button_fix(gen_settings)
         ################################################################################
         
         
@@ -4729,6 +4773,8 @@
 
     def apply(self):
         self.result = self.uom.get()
+        # macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+        macOS_button_fix(gen_settings)
         return 
 
 
@@ -5030,4 +5076,7 @@
     message_box("K40 Whisperer",LOAD_MSG)
 debug_message("Debuging is turned on.")
 
+# macOS Patch - Stephen Houser (stephenhouser@gmail.com)
+macOS_button_fix_enabled = True
+macOS_button_fix(root)
 root.mainloop()
--- K40_Whisperer-0.35_src/nano_library.py	2019-06-11 21:50:06.000000000 -0400
+++ nano_library.py	2019-09-08 12:37:02.000000000 -0400
@@ -281,10 +281,10 @@
                 FINISHED = True
                 break
             elif response == None:
-		msg = "Laser stopped responding after operation was complete."
+                msg = "Laser stopped responding after operation was complete."
                 update_gui(msg)
-		#raise Exception(msg)
-		FINISHED = True
+		        #raise Exception(msg)
+                FINISHED = True
             else: #assume: response == self.OK:
                 msg = "Waiting for the laser to finish."
                 update_gui(msg)
