--- K40_Whisperer-0.57_src/k40_whisperer.py	2021-03-16 19:22:44.000000000 -0700
+++ k40_whisperer.py	2021-03-23 10:37:50.000000000 -0700
@@ -5704,10 +5704,10 @@
 try:
     try:
         import tkFont
-        default_font = tkFont.nametofont("TkDefaultFont")
+        default_font = tkFont.nametofont("systemSystemFont")
     except:
         import tkinter.font
-        default_font = tkinter.font.nametofont("TkDefaultFont")
+        default_font = tkinter.font.nametofont("systemSystemFont")
 
     default_font.configure(size=9)
     default_font.configure(family='arial')
